ports/mynewt/src/memfault_platform_port.c

```diff
@@ -0,0 +1,142 @@
+//! @file
+//!
+//! Copyright (c) Memfault, Inc.
+//! See License.txt for details
+//!
+//! Glue layer between the Memfault SDK and the underlying platform
+//!
+//! TODO: Fill in FIXMEs below for your platform
+#include "sysinit/sysinit.h"
+#include "os/os.h"
+#include <bsp/bsp.h>
+
+#include "memfault/components.h"
+#include "memfault/ports/reboot_reason.h"
+
+#include <stdbool.h>
+
+void memfault_platform_get_device_info(sMemfaultDeviceInfo *info) {
+  // !FIXME: Populate with platform device information
+
+  // IMPORTANT: All strings returned in info must be constant
+  // or static as they will be used _after_ the function returns
+
+  // See https://mflt.io/version-nomenclature for more context
+  *info = (sMemfaultDeviceInfo) {
+    // An ID that uniquely identifies the device in your fleet
+    // (i.e serial number, mac addr, chip id, etc)
+    // Regular expression defining valid device serials: ^[-a-zA-Z0-9_]+$
+    .device_serial = NULL,
+     // A name to represent the firmware running on the MCU.
+    // (i.e "ble-fw", "main-fw", or a codename for your project)
+    .software_type = MYNEWT_VAL(MEMFAULT_DEVICE_INFO_SOFTWARE_TYPE),
+    // The version of the "software_type" currently running.
+    // "software_type" + "software_version" must uniquely represent
+    // a single binary
+    .software_version = MYNEWT_VAL(MEMFAULT_DEVICE_INFO_SOFTWARE_VERS),
+    // The revision of hardware for the device. This value must remain
+    // the same for a unique device.
+    // (i.e evt, dvt, pvt, or rev1, rev2, etc)
+    // Regular expression defining valid hardware versions: ^[-a-zA-Z0-9_\.\+]+$
+    .hardware_version = MYNEWT_VAL(MEMFAULT_DEVICE_INFO_HARDWARE_VERS),
+  };
+}
+//! Last function called after a coredump is saved. Should perform
+//! any final cleanup and then reset the device
+void memfault_platform_reboot(void) {
+  os_reboot(HAL_RESET_REQUESTED);
+  while (1) { } // unreachable
+}
+
+bool memfault_platform_time_get_current(sMemfaultCurrentTime *time) {
+  // !FIXME: If the device tracks real time, update 'unix_timestamp_secs' with seconds since epoch
+  // This will cause events logged by the SDK to be timestamped on the device rather than when they
+  // arrive on the server
+  *time = (sMemfaultCurrentTime) {
+    .type = kMemfaultCurrentTimeType_UnixEpochTimeSec,
```

Comment:
I think it may be better to use `os_time_get()` here.

Reply:
`os_time_get()` gives ticks, what is required here is `os_gettimeofday()`

Reply:
`os_gettimeofday` does not return system time in seconds, which is what is needed here to replace `kMemfaultCurrentTimeType_UnixEpochTimeSec1` instead current tick value can be converted to seconds by knowing the epoch and tick frequency.

Reply:
Discussed this seperately in slack and given the tick count is not actually the walltime and whats needed here is walltime so the `os_timeval` struct populated by `os_gettimeofday` is indeed whats needed here.